rust/osm2lanes/src/transform/lanes_to_tags.rs

```diff
@@ -174,11 +174,33 @@ pub fn lanes_to_tags(lanes: &[Lane], locale: &Locale, config: &LanesToTagsConfig
             .rev()
             .take_while(|lane| !lane.is_motor())
             .find(|lane| lane.is_bus());
-        match (left_bus_lane.is_some(), right_bus_lane.is_some()) {
-            (false, false) => {}
-            (true, false) => tags.checked_insert("busway:left", "lane")?,
-            (false, true) => tags.checked_insert("busway:right", "lane")?,
-            (true, true) => tags.checked_insert("busway:both", "lane")?,
+        if left_bus_lane.is_none()
+            && right_bus_lane.is_none()
+            && lanes.iter().any(|lane| lane.is_bus())
+        {
+            tags.checked_insert(
+                "bus:lanes",
+                lanes
+                    .iter()
+                    .map(|lane| if lane.is_bus() { "designated" } else { "" })
+                    .collect::<Vec<_>>()
```

Comment:
I don't know a better way to do this, just lamenting that it's not easier to build up the string without first making a full vector (and then doing `as_slice`). Might all be possible, but the error messages aren't simple if so. I miss Scala sometimes for reasons like this :\

Reply:
I always just assumed the implementation required a sequence in memory, but you made me go luck, and it doesn't appear to be the case: https://doc.rust-lang.org/src/alloc/slice.rs.html#776

Maybe my first PR to rust can be to try and add join to the Iterator trait...